modules/ROOT/pages/db/db-connector-insert-update-delete.adoc

```diff
@@ -3,48 +3,157 @@ ifndef::env-site,env-github[]
+    INSERT INTO electronic(name, description, price, discount)
+    VALUES ('Coffee Machine', 'Model: XYZ99. Uses small size capsules.', 120, 5)
   </db:sql>
-  <db:input-parameters>
-    #[
-    {'description' : payload}
-    ]
-  </db:input-parameters>
 </db:insert>
-view rawinsert.xml hosted with ❤ by GitHub
+----
+
+Note that the *id* parameter is not passed in the query because it is an
+autoincrement key that is generated automatically by the database.
+
+[NOTE]
+=============
+To benefit from the advantages of
+xref:db/db-connector-query.adoc#_use_input_parameters_to_protect_database_queries[input parameters],
+the equivalent query should be written as follows:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig">
+    <db:sql>
+        INSERT INTO electronic(name, description, price, discount)
+        VALUES (:name, :description, :price, :discount)
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+=============
+
+The `insert` operation will output a `Statement Result` object that contains 2 fields: a _affectedRows_
+integer that indicates how many rows were affected by the query, and a _generatedKeys_ map
+which contains the autogenerated keys. Given this example, the output is as follows:
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+The _generatedKeys_ map is empty, although a row is successfully added to the table.
+This is because this information is disabled by default in order to avoid large overheads.
+To activate the information and generate keys, add the `autoGenerateKeys` parameter, and then set it to true:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true">
+   <db:sql>
+       INSERT INTO electronic(name, description, price, discount)
+       VALUES (:name, :description, :price, :discount)
+   </db:sql>
+   <db:input-parameters>#[{
+       name: 'Coffee Machine',
+       description: 'Model:XYZ99. Uses small size capsules.',
+       price: 120,
+       discount: 5}]
+   </db:input-parameters>
+</db:insert>
+----
+
+If the statement generates multiple keys per affected row, you can specify which columns
+should be returned by setting the parameter `autoGeneratedKeysColumnNames` to a list that contains
+the desired column names. For example, to ensure only the _id_ column is
+returned, set the parameter to a DataWeave list with a single string element:
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true" autoGeneratedKeysColumnNames="#[['id']]">
+    ...
+</db:insert>
+----
+
+== Update Example
+
+Given the table that is defined in the `insert` example, the operation to update the *electronic* table
+to set the discount to 10 percent for all items that have a price above 100 is as follows:
```

Comment:
Should we call this a *price* value? "Price above 100" just sounds odd to me....

Reply:
Yes. Added.